app/models/manageiq/providers/ansible_tower/provider.rb

```diff
@@ -5,6 +5,14 @@ class ManageIQ::Providers::AnsibleTower::Provider < ::Provider
           :dependent   => :destroy,
           :autosave    => true
 
+  has_many :endpoints, :as => :resource, :dependent => :destroy, :autosave => true
+
+  delegate :url,
+           :url=,
+           :to => :default_endpoint
+
+  virtual_column :url, :type => :string, :uses => :endpoints
```

Comment:
The reason I added `has_many :endpoints` directly to Ansible and Foreman, was otherwise we were getting the following error
```ruby
ActiveRecord::HasManyThroughCantAssociateThroughHasOneOrManyReflection:
Cannot modify association 'ManageIQ::Providers::Foreman::Provider#endpoints'
because the source reflection class 'Endpoint' is associated to
'ExtManagementSystem' via :has_many.
```
I'm not happy about it, but I wasn't really sure how else to fix this error.